### PR TITLE
Closing exception print writer

### PR DESCRIPTION
**Pull request by yurir**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

Added closing for exception print writer

### CODE REVIEW

**Comment by gemelen** [comment]
@yurir I suggest to rewrite this using [arm4s](https://github.com/tmoschou/arm4s), so both StringWriter and PrintWriter would be closed.
This would be like:
```
def getStringStackTrace: String = {
      for { 
        sw <- new StringWriter()
        pw <- new PrintWriter(sw)
        e.printStackTrace(pw)
    } yield (sw.toString)
```

**Comment by yurir** [comment]
@gemelen interesting library. However, is it worth it to add this dependency for such corner case? BTW, the project already has scala-arm.

**Comment by gemelen** [comment]
I personally prefer arm4s, but it's better to use one which is already present in project (cause they both provide kinda similar approaches for resource management), so feel free to use scala-arm for this and other cases.

File: infra/src/main/scala/com/aol/one/dwh/infra/util/ExceptionPrinter.scala [hunk 1]
```diff
@@ -10,14 +10,22 @@ package com.aol.one.dwh.infra.util
...
+
 trait ExceptionPrinter {
 
   implicit class ExceptionOpts(e: Throwable) {
     def getStringStackTrace: String = {
-      val sw = new StringWriter()
-      val pw = new PrintWriter(sw)
-      e.printStackTrace(pw)
-      sw.toString
+      var stringWriter: Option[StringWriter] = None
```
Comment (gemelen) [inline infra/src/main/scala/com/aol/one/dwh/infra/util/ExceptionPrinter.scala]: @yurir you do not need any var for intermediate storage. Also this for is not a cycle statement but an other form for flatMap function, so you may expect that vale defined and initialized inside of first brackets, will be available in yield (second brackets). See [this doc](https://docs.scala-lang.org/tutorials/FAQ/yield.html) for example.

Comment (yurir) [reply infra/src/main/scala/com/aol/one/dwh/infra/util/ExceptionPrinter.scala]: @gemelen  can't make it to compile in the way you suggested. Any changes cause 'sw' or 'e' to be unreachable.

Comment (gemelen) [reply infra/src/main/scala/com/aol/one/dwh/infra/util/ExceptionPrinter.scala]: @yurir please, find an example [here](http://jsuereth.com/scala-arm/usage.html).
Code would be like this
```
    def getStringStackTrace: String = {
      val sw = managed(new StringWriter())
      val pw = sw map (new PrintWriter(_))

      pw map (e.printStackTrace(_))
      sw toString()
    }

```

Comment (yurir) [reply infra/src/main/scala/com/aol/one/dwh/infra/util/ExceptionPrinter.scala]: @gemelen it compiles, but doesn't work...

Comment (yurir) [reply infra/src/main/scala/com/aol/one/dwh/infra/util/ExceptionPrinter.scala]: @gemelen , i suggest that i merge the PR now, and you rewrite this piece of code when you have time :-)

File: project/build.properties [hunk 2]
```diff
@@ -1 +1 @@
-sbt.version = 1.2.7
+sbt.version = 1.2.8
```
Comment (alexey-kotlyar) [inline project/build.properties]: please remove this change from the change set, as we are handling only exception printwriter in current scope

File: infra/src/main/scala/com/aol/one/dwh/infra/util/ExceptionPrinter.scala (hunk 1 above)
Comment (gemelen) [reply infra/src/main/scala/com/aol/one/dwh/infra/util/ExceptionPrinter.scala]: @yurir I'm personally against of merging it as is (with either Java-style code or intermediate solution).
I believe it is not a critical issue with Bandarlog that is to be solved with that changeset, so we could implement it in a right way without a much hustle.

File: README.md [hunk 3]
```diff
@@ -406,9 +406,9 @@ Bandar-Log is released under the Apache License, Version 2.0
 
 ## Credits
  * [Alexey Lipodat](mailto:a.lipodat@gmail.com)
- * [Denis Pyshev](mailto:denys.pyshev@verizonmedia.com )
```
Comment (alexey-kotlyar) [inline README.md]: mm, those changes are already in master, aren't they?

File: project/build.properties (hunk 2 above)
Comment (yurir) [reply project/build.properties]: @alexey-kotlyar done

@gemelen i reduced lines of code number

File: README.md (hunk 3 above)
Comment (yurir) [reply README.md]: @alexey-kotlyar currently in master appears denys's sigma mail address

File: infra/src/main/scala/com/aol/one/dwh/infra/util/ExceptionPrinter.scala (hunk 1 above)
Comment (yurir) [reply infra/src/main/scala/com/aol/one/dwh/infra/util/ExceptionPrinter.scala]: @gemelen i reduced lines of code number. please have a look.

**Comment by gemelen** [comment]
@yurir @alexey-kotlyar please have a look.
@yurir just to make it clear, this method converts list of StackTraceElement (ie set of strings) of an exception into one string, has nothing to do with exception message or other its members and is used in realtime-processing applications too, so you could refer for use case their in EnrichmentActionImpl

**Comment by yurir** [comment]
@gemelen LGTM. Thanks. I can't approve the PR, since i initially created it (github feature).

**Pull request merged**